okhttp/src/jvmMain/kotlin/okhttp3/ConnectionPool.kt

```diff
@@ -33,30 +33,45 @@ import okhttp3.internal.connection.RealConnectionPool
 class ConnectionPool internal constructor(
   internal val delegate: RealConnectionPool
 ) {
-  constructor(
-    maxIdleConnections: Int,
-    keepAliveDuration: Long,
-    timeUnit: TimeUnit
+  internal constructor(
+    maxIdleConnections: Int = 5,
+    keepAliveDuration: Long = 5,
+    timeUnit: TimeUnit = TimeUnit.MINUTES,
+    taskRunner: TaskRunner = TaskRunner.INSTANCE,
+    connectionListener: ConnectionListener = ConnectionListener.NONE,
   ) : this(RealConnectionPool(
-      taskRunner = TaskRunner.INSTANCE,
+      taskRunner = taskRunner,
       maxIdleConnections = maxIdleConnections,
       keepAliveDuration = keepAliveDuration,
       timeUnit = timeUnit,
-      connectionListener = ConnectionListener.NONE
+      connectionListener = connectionListener
   ))
 
   constructor(
+    connectionListener: ConnectionListener = ConnectionListener.NONE,
```

Comment:
nit: surprising that the new parameter is first?

Reply:
I can reorder. It seemed right to me. But should we go full Builder mode on it?

Reply:
Follow up https://github.com/square/okhttp/issues/7631

Reply:
No Builder.